specs/data_structures.md

```diff
@@ -320,31 +286,178 @@ A proof into an SMT is structured as:
 
 # Erasure Coding
 
+In order to enable trust-minimized light clients (i.e. light clients that do not rely on an honest majority of validating state assumption), it is critical that light clients can determine whether the data in each block is _available_ or not, without downloading the whole block itself. The technique used here was formally described in the paper [Fraud and Data Availability Proofs: Maximising Light Client Security and Scaling Blockchains with Dishonest Majorities](https://arxiv.org/abs/1809.09044).
+
+The remainder of the subsections below specify the [2D Reed-Solomon erasure coding scheme](#2d-reed-solomon-encoding-scheme) used, along with the format of [shares](#share) and how [available data](#available-data) is arranged into shares.
+
+## Reed-Solomon Erasure Coding
+
+Reed-Solomon erasure coding is used as the underlying coding scheme. The parameters are:
+- 16-bit Galois field
+- 1:1 original:parity pieces
```

Comment:
Technically you can have an erasure coding where you need e.g 80% of the total shares to recover. This parameter is equivalent to saying "50%."

Reply:
Ok, I see. This probably has a name in the erasure coding world. Best to use that word instead. So I guess the param is 2 (meaning you can loose half the data / need half the data to recover).

Reply:
What you are describing seems to be called _erasure efficiency_.
> The fraction k’/k, where k’ denotes the number of symbols required for recovery, is called reception efficiency.
So in the general sense the parameter would be the the desired erasure efficiency (here 1/2).

In the context of reed solomon this is simply defined by the number of check symbols (or parity pieces). So the parameter is the number of parity symbols:

> By adding t check symbols to the data, a Reed–Solomon code can detect (but not correct) any combination of up to and including t erroneous symbols, OR locate and correct up to and including ⌊t/2⌋ erroneous symbols at unknown locations. As an erasure code, it can correct up to and including t erasures at locations that are known and provided to the algorithm, or it can detect and correct combinations of errors and erasures.

If we fix one half, the only actual param is the number of original data symbols (and maybe the symbol size).